### DISCUSSION TITLE
CI/CD approach for shared services

### DISCUSSION THREAD

**Discussion by alex5207** [aws-samples/aws-saas-factory-ref-solution-serverless-saas#40] [Q&A]
Great reference architecture introducing a lot of key concept - Thanks!

The repo uses `CodePipeline` for deployment of the application services and tenant specific infrastructure.

I am wondering whether there is a specific reason for not re-building the shared services as well? Is there anything one should be aware of when making updates to the shared services as opposed to the application services? E.g the `auth_manager.py` layer responsible for constructing policies needs to be updated if introducing new application services.

How should one go about extending the ci/cd approach to include these parts of the application?

Thanks

**Comment by anubhavynr** [comment]
Thanks... Yes, you would normally create a separate CI\CD pipeline for shared services as well. Reason we didn't spent time doing this here because we wanted to focus on the application plane multi-tenant pipeline, which is more nuanced in case of SaaS based applications.

**Comment by alex5207** [comment]
Sure, makes perfect sense. In case of implementing the CI/CD pipeline for the shared services, what kind of considerations would be important? I guess commits to the repo should somehow be segregated such that only the relevant pipeline is triggered? Any guidance of links to resources of how to approach this would be helpful!

**Reply by anubhavynr** [reply]
sorry for the delay in response. Yes, normally a separate code repository should be created for shared services that can also contain the CI CD pipeline to take care of the deployment.
